### ISSUE TITLE
pre-trained model download problem

### ISSUE THREAD

**Issue by sandyaa0313** [haowang1992/DSN#1]
Is it possible to provide a download link on google drive? I can't download it on baidu. Thank you!

**Comment by zaiquanyang** [comment]
> Is it possible to provide a download link on google drive? I can't download it on baidu. Thank you!

Do you have reproduced it? I meet some problems and be eager for answers.

**Comment by sandyaa0313** [comment]
Yes, I've reproduced it. Do you fix your problems?

**Comment by 1090h2400** [comment]
memory.add_entries(features[:tag_zeros.size(0)].detach(), label_s.detach(),
                           features[tag_zeros.size(0):tag_all.size(0)].detach(), label_i.detach())
        loss_memory = 1.0 - memory.memory_loss(features[tag_zeros.size(0):tag_all.size(0)].detach(), label_i.detach())
.detach()  causes loss_memory.grad to be none.  It's not being used? （loss_memory ）.

**Comment by 1090h2400** [comment]
> Yes, I've reproduced it. Do you fix your problems?
  Hi，Do you know why detach is used ？ And .detach() causes loss_memory.grad to be none
    memory.add_entries(features[:tag_zeros.size(0)].detach(), label_s.detach(),
                       features[tag_zeros.size(0):tag_all.size(0)].detach(), label_i.detach())
    loss_memory = 1.0 - memory.memory_loss(features[tag_zeros.size(0):tag_all.size(0)].detach(), label_i.detach())

**Comment by Qingzhang1994** [comment]
Would you like to ask how to visualize top-K retrieval results，thank。
![image](https://user-images.githubusercontent.com/48178899/190539165-58ea6aa4-5a36-4184-a4eb-4af48b883d8d.png)
